src/main/java/com/Elessar/app/client/MyClient.java

```diff
@@ -22,60 +25,90 @@
+                                                                           .append(LOGON).toString());
 
-            final HttpResponse response = httpClient.post(new URL(hostURL + "/logon"), logonRequest.build());
-            return LogonResponse.parseFrom(response.getContent());
+        final LogonRequest.Builder logonRequest = LogonRequest.newBuilder().setName(userName)
+                                                                           .setPassword(hash(password))
+                                                                           .setPort(clientPort);
+
+        final HttpResponse response = httpClient.post(new URL(hostURL + "/logon"), logonRequest.build());
+
+        metric.timerStop();
+
+        return LogonResponse.parseFrom(response.getContent());
     }
 
+
     public LogoffResponse logOff(String userName) throws Exception {
-            final LogoffRequest.Builder logoffRequest= LogoffRequest.newBuilder().setName(userName);
+        final Metric metric = metricManager.newMetric(new StringBuilder().append(CLIENT).append(".")
+                                                                           .append(LOGOFF).toString());
+
+        final LogoffRequest.Builder logoffRequest= LogoffRequest.newBuilder().setName(userName);
 
-            final HttpResponse response = httpClient.post(new URL(hostURL + "/logoff"), logoffRequest.build());
-            return LogoffResponse.parseFrom(response.getContent());
+        final HttpResponse response = httpClient.post(new URL(hostURL + "/logoff"), logoffRequest.build());
+
+        metric.timerStop();
+
+        return LogoffResponse.parseFrom(response.getContent());
     }
 
+
     public RegistrationResponse register (String userName,
                                           String password,
                                           String email,
                                           String phoneNumber) throws Exception {
+        final Metric metric = metricManager.newMetric(new StringBuilder().append(CLIENT).append(".")
+                                                                           .append(REGISTER).toString());
+
         final RegistrationRequest.Builder registerRequest = RegistrationRequest.newBuilder().setName(userName)
                                                                                             .setPassword(hash(password))
-                                                                                            .setEmail(email);
-        if (phoneNumber != null) {
-            registerRequest.setPhoneNumber(phoneNumber);
-        }
+                                                                                            .setEmail(email)
+                                                                                            .setPhoneNumber(phoneNumber);
 
         final HttpResponse response = httpClient.post(new URL(hostURL + "/register"), registerRequest.build());
+
+        metric.timerStop();
```

Comment:
just point out that if we should really make "metric.timerStop" the last line of each block of code we want to measure performance.